backend/cfg/merge_straightline_blocks.ml

```diff
@@ -71,7 +71,16 @@ let rec merge_blocks (modified : bool) (cfg_with_layout : Cfg_with_layout.t) : b
              b1_block.can_raise_interproc <- b1_block.can_raise_interproc || b2_block.can_raise_interproc;
              (* modify b2 *)
              b2_block.predecessors <- Label.Set.empty;
-             Disconnect_block.disconnect cfg_with_layout b2_label;
+             Label.Set.iter
```

Comment:
Why not calling `Disconnect_blocks` here? Is it the trap handlers check in `Disconnect_blocks` that does not pass?

Reply:
I don't think one can try to disconnect a trap handler block given
the shapes of the CFGs currently produced by the middle-end; the
issue was with one of the `assert` if memory serves.